### PR TITLE
fix(amf): Configured policy ambr and QFI related fix

### PR DESCRIPTION
**Pull request by Sathyaj27**
Signed-off-by: Sathyaj27 <sathya.jayadev@wavelabs.ai>

## Summary
Configured policy ambr values were not getting reflected in amf. Added required fix by filling with correct values

## Test Plan
Tested on Tera Vm
UT passed
MicrosoftTeams-image (4)
newoutput

## Additional Information

- [ ] This change is not backwards-breaking

zenhub # 11866

### CODE REVIEW

File: lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 1]
```diff
@@ -119,14 +118,19 @@ Status AmfServiceImpl::SetSmfSessionContext(
...
   itti_msg.session_ambr.uplink_unit_type = req_m5g.subscribed_qos().br_unit();
-  itti_msg.session_ambr.uplink_units = req_m5g.subscribed_qos().apn_ambr_ul();
-
   itti_msg.session_ambr.downlink_unit_type = req_m5g.subscribed_qos().br_unit();
-  itti_msg.session_ambr.downlink_units = req_m5g.subscribed_qos().apn_ambr_dl();
-
-  // authorized qos profile
-  itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
-      req_m5g.subscribed_qos().qos_class_id();
+  if (!req_m5g.qos().max_req_bw_ul() && !req_m5g.qos().max_req_bw_dl()) {
```
Comment (panyogesh) [inline lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp:123]: Put some comments for clarity.

**Comment by panyogesh** [comment]
Just a minor comment. Rest looks fine.

File: lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp [hunk 2]
```diff
@@ -119,14 +118,22 @@ Status AmfServiceImpl::SetSmfSessionContext(
...
+    itti_msg.session_ambr.downlink_units =
+        req_m5g.subscribed_qos().apn_ambr_dl();
+    itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
+        req_m5g.subscribed_qos().qos_class_id();
+  } else {
+    // Policy ambr value if policy attached
+    itti_msg.session_ambr.uplink_units = req_m5g.qos().max_req_bw_ul();
+    itti_msg.session_ambr.downlink_units = req_m5g.qos().max_req_bw_dl();
+    itti_msg.qos_list.qos_flow_req_item.qos_flow_identifier =
+        req_m5g.qos().qci();
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp]: can you add more details on how is this qos policy configured?

Comment (Sathyaj27) [reply lte/gateway/c/core/oai/tasks/grpc_service/AmfServiceImpl.cpp]: It is configured through the nms, We should have an active policy.
Attaching a screen shot where policy is not attached:
MicrosoftTeams-image (5)

**Review by pruthvihebbani — APPROVED** [review]

**Comment by pruthvihebbani** [comment]
@Sathyaj27 Please rebase with latest master.

**Comment by Sathyaj27** [comment]
@pruthvihebbani  Rebased Changes

**Comment by pruthvihebbani** [comment]
lte test job i.e. mobilityd/tests is failing. Can you please check

**Comment by Sathyaj27** [comment]
@pruthvihebbani : This is unrelated to my changes and its known issue https://magmacore.slack.com/archives/C01MDUQDFC3/p1648222335621339

**Pull request merged**